backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/allocation/transfer/HUTransformService.java

```diff
@@ -1396,9 +1401,12 @@ private void addCUsToTU(
 	}
 
 	@NonNull
-	public HuId extractToTopLevelByQRCode(@NonNull final HuId huId, @NonNull final HUQRCode huQRCode)
+	public HuId extractToTopLevelByQRCode(@NonNull final HuId huId, @Nullable final HUQRCode huQRCode)
```

Comment:
IMHO u can now rename `extractToTopLevelByQRCode` to `extractToTopLevel` because is not necessary by qr code